### ISSUE TITLE
Statusbar should show "Nearest" error

### ISSUE THREAD

**Issue by Saevon** [SublimeLinter/SublimeLinter#1034]
While it is nice to see the error only under your cursor this was only a problem when the errors got too long. Usually I have a single error for a line, and when my cursor is on the line I don't want to go to a specific character to see the error. This is especially pronounced when you have the errors panel open, as I'm not usually on the first character of the error for things like "Import Errors" so the panel won't highlight it, nor jump to it.

I'd suggest one of the following:

1. Adding an option for the previous version (show all errors on line) (compatibility)
2. Showing the "closest" error to the current cursor on the same line (Best UI)
3. If there are multiple errors on the line (and your cursor isn't on an error) then show just the first one (Easiest Implementation, I believe)

**Comment by braver** [comment]
> This is especially pronounced when you have the errors panel open, as I'm not usually on the first character of the error for things like "Import Errors" so the panel won't highlight it, nor jump to it.

We haven't released #1024 yet, but the panel will tell you where you are and what the nearest errors are.

The status bar should provide contextual information that is easy to understand. If there are multiple errors on one line and the cursor is right between two of them, and then we display the "nearest", which one is that actually?

Showing all errors on the line usually effectively only displays the first one, the others will be clipped. Doing so will also fully claim the status bar.

With #1024 I think we have the most optimal mix of features. But we'll have to see if people can get used to it, and/or if we need to change something.

**Comment by kaste** [comment]
I propose that we should never display the error in the status bar. It is clumsy, I can't read it (weak eyes) even though I already increased the font size and spacing of my status bar. Also, the status bar is just not approriate for displaying this kind of information. I personally need this near the cursor.

As soon as we disable/remove this. We could have a plugin which provides this functionality in a clever or very opinionated way.

**Comment by braver** [comment]
Oh yeah, I remembered that we also have the "line report" that you can always call to read all the errors on the current line.

I'm super used to glancing at the status bar for info on the current error. But I can totally get behind moving this to a plugin and/or toggling with a setting.

**Comment by Saevon** [comment]
the problem with #1024 is that it requires the panel view, and the only time I would use that is when I have many errors across my project/file.
When I'm doing normal coding just one or two errors would pop up, and I would fix them quickly. Not needing to have the panel open (screen real-estate) NOR needing to move my cursor to the exact error position means I can fix the errors quickly as I code with the least interruption.

When there are multiple errors I don't mind that the workflow doesn't work... I can use the panel to solve that (I used to use the old one for it, so the new one will work fine still)

"Line Report" is basically the new panel isn't it? (Not sure I see that command)

@kaste I'd rather not remove it just cause some people can't read it... the status bar being hard to read is a Sublime issue, and should be solvable by Sublime color schemes (I would love if Sublime could change the font-size of it too)

TL;DR A Setting is fine for me too

**Comment by braver** [comment]
> "Line Report" is basically the new panel isn't it? 

@Saevon use the "show line errors" command.

Wouldn't jumping to the error position (via the goto command) be beneficial to quickly fixing the error?

I'm not against adding an option here, we just need to understand it well. For instance, if we show the "nearest", what is actually the nearest (how do we determine it) and how do you know what that nearest error is? I think I get how it can be handy if it works well, but the UX isn't clear to me right now. A PR we can play with to learn more would definitely help here.

**Comment by Saevon** [comment]
ah, I meant the nearest on the **same line**, (AKA I like that you don't show **all** the errors now, but now I get **none** most of the time).

My use-case goes like this:

1. I'm editing code normally (no errors/warnings exist at all)
2. a new problem pops up on the line I'm editing (or one or two lines up)
3. (I'm still writing new code) I navigate to a line with an error while fixing/adding unrelated stuff, and take the moment to fix the error
4. Glance down at the "status bar" so I can quickly fix the problem... But now I need to navigate the cursor to the exact highlighted section

hopefully thats clear

**Comment by kaste** [comment]
Sure, I meant separate it from core and do it in a plugin.

**Comment by Saevon** [comment]
@kaste, considering how small of a change this is, a setting seems like a better answer than plugin

**Comment by kaste** [comment]
You actually propose to show a nearest error. If you just mean a enabled/disabled setting, all event listeners are registered on load time in Sublime, so you cannot avoid the or some of the computational work. Currently this view-plugin is very minimal. Any sophisticated UI relative quickly grows.

**Comment by Saevon** [comment]
It already has to recheck if there's an error under the cursor, (assuming that uses some error map based on file positions), it could just use line numbers to do the lookup instead.

Even if the statusbar isn't the one showing that information, if you have the error-panel open that too needs to update as you move your cursor, thus no extra computational work for keeping it in

I'll see if I can get a PR in then, thanks
